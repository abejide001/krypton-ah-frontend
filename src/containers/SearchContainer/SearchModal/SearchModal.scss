i.icon {
  float: right;
  text-align: center;
  cursor: pointer;
}
.ui.modal > .header {
  margin: 0 0 2rem 0;
  border: none;
}

.ui.input.search {
  > input {
    text-align: center;
    font-family: inherit;
    font-size: 2rem;
    border: {
      top: 0;
      left: 0;
      right: 0;
      bottom: 0.2rem solid rgba(34, 36, 38, 0.15);
    }
  }
}

.ui.input.innerSearch {
  > input {
    max-width: 50%;
    margin: 0 auto;
  }
}

.ui.fullscreen.modal {
  width: 100% !important;
}
.radioMargin {
  margin-right: 0.7rem;
  font-weight: 400;
}

.ui.segment.radioCenter {
  text-align: center;
}
