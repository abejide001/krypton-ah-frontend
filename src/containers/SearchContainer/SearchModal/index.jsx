import React, { Component, Fragment } from 'react';
import { Modal, Icon, Input, Segment, Radio } from 'semantic-ui-react';
import { Redirect } from 'react-router-dom';
import './SearchModal.scss';

class SearchModal extends Component {
  constructor(props) {
    super(props);
    this.state = {
      article: '',
      open: false,
      search: false,
      searchOption: '',
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleRadioChange = this.handleRadioChange.bind(this);
  }

  show = size => () => this.setState({ size, open: true });

  close = () => this.setState({ open: false });

  handleRadioChange(event, { value }) {
    this.setState({ searchOption: value });
  }

  handleChange(event) {
    const { name, value } = event.target;
    this.setState(state => ({
      ...state,
      [name]: value
    }));
  }

  handleSubmit(event) {
    event.preventDefault();
    this.setState({
      search: true
    });
  }

  render() {
    const { open, size, article, searchOption } = this.state;
    if (searchOption) {
      if (this.state.search) { return <Redirect to={`/search/${searchOption}?value=${article}`} />; }
    } else if (this.state.search) { return <Redirect to={`/search/${searchOption}?value=${article}`} />; }
    return (
      <Fragment>
        <Icon size="big" name="search" onClick={this.show('fullscreen')} />
        <Modal size={size} open={open} onClose={this.close}>
          <Modal.Header className="headerBorder">
            <Icon name="close" onClick={this.close} />
          </Modal.Header>
          <Modal.Content>
            <form onSubmit={this.handleSubmit}>
              <Input type="text" name="article" value={article} placeholder="Type and hit Enter to search..." className="search" onChange={this.handleChange} />
              <Segment>
                <Radio label="Author" name="searchOption" value="author" className="radioMargin" checked={this.state.searchOption === 'author'} onChange={this.handleRadioChange} />
                <Radio label="Tag" name="searchOption" value="tag" className="radioMargin" checked={this.state.searchOption === 'tag'} onChange={this.handleRadioChange} />
                <Radio label="Title" name="searchOption" value="title" className="radioMargin" checked={this.state.searchOption === 'title'} onChange={this.handleRadioChange} />
              </Segment>
            </form>
          </Modal.Content>
        </Modal>
      </Fragment>
    );
  }
}

export default SearchModal;
