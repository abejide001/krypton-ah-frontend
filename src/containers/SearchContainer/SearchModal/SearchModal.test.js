import React from 'react';
import { shallow } from 'enzyme';
import SearchModal from '.';

describe('Search component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<SearchModal />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should change the state after change in the input value', () => {
    const newValue = 'search';
    const wrapper = shallow(<SearchModal />);
    wrapper.find('Input[type="text"]').simulate('change', {
      target: { name: newValue }
    });
    wrapper.update();
    expect(wrapper.find('Input[type="text"]').prop('name')).toContain('article');
  });

  it('should change the state after change in radio button value', () => {
    const newValue = 'search';
    const wrapper = shallow(<SearchModal />);
    wrapper.instance().handleRadioChange({}, { value: newValue });
    expect(wrapper.state().searchOption).toBe(newValue);
  });

  it('should simulate open on modal', () => {
    const wrapper = shallow(<SearchModal />);
    wrapper.instance().show();
    expect(wrapper).toBeDefined();
  });

  it('should simulate close on modal', () => {
    const wrapper = shallow(<SearchModal />);
    wrapper.instance().close();
    expect(wrapper).toBeDefined();
  });

  it('should simulate on submit event', () => {
    const wrapper = shallow(<SearchModal />);
    wrapper.find('form').simulate('submit', {
      preventDefault: jest.fn()
    });
    expect(wrapper.find('Redirect')).toBeDefined();
  });
});
