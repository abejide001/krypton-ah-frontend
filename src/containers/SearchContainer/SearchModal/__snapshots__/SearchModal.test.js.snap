// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component should match snapshot 1`] = `
<Fragment>
  <Icon
    as="i"
    name="search"
    onClick={[Function]}
    size="big"
  />
  <Modal
    centered={true}
    closeOnDimmerClick={true}
    closeOnDocumentClick={false}
    dimmer={true}
    eventPool="Modal"
    onClose={[Function]}
    open={false}
  >
    <ModalHeader
      className="headerBorder"
    >
      <Icon
        as="i"
        name="close"
        onClick={[Function]}
      />
    </ModalHeader>
    <ModalContent>
      <form
        onSubmit={[Function]}
      >
        <Input
          className="search"
          name="article"
          onChange={[Function]}
          placeholder="Type and hit Enter to search..."
          type="text"
          value=""
        />
        <Segment>
          <Radio
            checked={false}
            className="radioMargin"
            label="Author"
            name="searchOption"
            onChange={[Function]}
            type="radio"
            value="author"
          />
          <Radio
            checked={false}
            className="radioMargin"
            label="Tag"
            name="searchOption"
            onChange={[Function]}
            type="radio"
            value="tag"
          />
          <Radio
            checked={false}
            className="radioMargin"
            label="Title"
            name="searchOption"
            onChange={[Function]}
            type="radio"
            value="title"
          />
        </Segment>
      </form>
    </ModalContent>
  </Modal>
</Fragment>
`;
